client/app/bundles/course/video/components/tables/VideoTable.tsx

```diff
@@ -40,7 +40,19 @@ const VideoTable: FC<Props> = (props) => {
     return <Note message={intl.formatMessage(translations.noVideo)} />;
   }
 
-  videos.sort((a, b) => a.title.localeCompare(b.title));
+  const videoSortMethodByDate = (
+    val1: VideoListData,
+    val2: VideoListData,
+  ): number => {
+    const post1 = val1.startTimeInfo;
+    const post2 = val2.startTimeInfo;
+
+    const date1 = new Date(post1.referenceTime!);
```

Comment:
Dont use `!`

Reply:
actually we can. I tried to make a new video without specifying the "start_at" date, the request for creating one is considered a Bad Request and hence rejected. I think this also means I need to revise the Yup requirement in this Video object, to not allow the "Starts At" to be null

Reply:
You cannot assume that way based on the input. There could be some data processing due to business logic within the app. You need to check the source of data (ie referenceTime)

Reply:
I understand. I need to take a look again on how the referenceTime is being defined. In the current commit (I just pushed it), I included the consideration of referenceTime being undefined as well